### PR TITLE
Add support for configuring disk size in parallels-ipsw builder

### LINKED ISSUE #111: Add support for configuring disk size in parallels-ipsw builder
#### Community Note

Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request.
If you are interested in working on this issue or have submitted a pull request, please leave a comment.

#### Description

Currently, the parallels-ipsw builder does not configure the VM's disk size. This results in the box size defaulting to 128 GB. Can an option be added to configure the size?

#### Use Case(s)

A clean macOS Sonoma install consumes about 23 GB of disk space. Reserving room for data, a disk size of 30-40GB would be relatively more manageable than the default 128GB.

#### Potential configuration

The same implementation as the parallels-iso builder will work here. 

1. Create VM and pass `--no-hdd` to `prlctl`
2. Call `stepCreateDisk` to create the new disk. `DiskSize` is already defined in the ipsw builder.

#### Contribute
If adding this feature makes sense, I'm happy to submit a PR.

### PR DESCRIPTION
**Pull request by johnbyrneio**
Adds support for specifying `disk_size` when using the ipsw builder. 

Follows the same implementation as the iso builder, except that macOS on Apple Silicon only supports `disk_type`: `plain`, and `iface`: `sata`.  For that reason, I hard-coded those values in `step_create_disk.go`.

Closes #111

### CODE REVIEW

**Comment by cjlapao** [comment]
@johnbyrneio this is failing on the build step, please have a look so we can merge this, thanks

**Review by cjlapao — APPROVED** [review]

**Review by bineesh-n — APPROVED** [review]

**Comment by cjlapao** [comment]
@johnbyrneio We are still waiting for the fix in the Go Validate step before we can merge this. because this is a fork we are unable to fix it from our side. could you please have a look at this?

**Comment by cjlapao** [comment]
@johnbyrneio to fix the issue just run `make install-packer-sdc` and then `make generate` this should generate the missing files. It has to do with missing documentation that is auto generated

**Comment by johnbyrneio** [comment]
@cjlapao I just worked that out. Should be good now.

**Review by cjlapao — APPROVED** [review]

**Review by bineesh-n — APPROVED** [review]

**Comment by cjlapao** [comment]
@johnbyrneio thanks for the contribution, merged it

**Pull request merged**
